From 6870acd0c3e7b4b0c3f9abbbc172ecb31c0e718e Mon Sep 17 00:00:00 2001
From: Ron Eldor <Ron.Eldor@arm.com>
Date: Thu, 12 Jul 2018 13:39:17 +0300
Subject: [PATCH 25/43] Change tmp buffer size to be as closest to orig

When `MBEDTLS_SHA512_C` is defined, the prf function uses Sha384.
In this case, the tmp buffer should be at least 131 bytes becuase:
`md_len` is 48, `strlen("EXTRACTOR-dtls_srtp")` is 19 and `rlen`
is 64. This sums to 131, and original size of 128 is not enough.
---
 library/ssl_tls.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/library/ssl_tls.c b/library/ssl_tls.c
index dad9e3f17..6f35ac6dc 100644
--- a/library/ssl_tls.c
+++ b/library/ssl_tls.c
@@ -498,7 +498,7 @@ static int tls_prf_generic( mbedtls_md_type_t md_type,
 {
     size_t nb;
     size_t i, j, k, md_len;
-    unsigned char tmp[144];
+    unsigned char tmp[132];
     unsigned char h_i[MBEDTLS_MD_MAX_SIZE];
     const mbedtls_md_info_t *md_info;
     mbedtls_md_context_t md_ctx;
-- 
2.17.1

